### PR TITLE
Add Select transformer for replacing clockshift and sampler

### LINKED ISSUE #257: Replace Sampler and ClockShift by a more intuitive transformer
**What is your feature request related to?**
* Module request

**Describe the solution you'd like**
Enable something like that instead of using Sampler and ClockShifts which is confusing.

```Select(start=-23, stop=0, step=1)
Select(start=0, stop=24, step=1)
Select(-24)
Select(start=-167, stop=0, step=24)```

### PR DESCRIPTION
**Pull request by benHeid**
## Related Issues
closes #257 
## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos)
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Did you update the CHANGELOG

## PR review
Anyone in the community is free to review the PR once the tests have passed.

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones to the PR so it can be classified

### CODE REVIEW

File: examples/example.py [hunk 1]
```diff
@@ -39,20 +40,17 @@
     scale_power_statistics = power_scaler(x=imputer_power_statistics)
 
     # Create lagged time series to later be used as regressors
-    shift_power_statistics = ClockShift(lag=1, name="ClockShift_Lag1"
-                                        )(x=scale_power_statistics)
-    shift_power_statistics2 = ClockShift(lag=2, name="ClockShift_Lag2"
-                                         )(x=scale_power_statistics)
+    lag_features = Select(start=-2, stop=0, step=1)(x=scale_power_statistics)
```
Comment (olineumann) [inline examples/example.py:43]: Maybe we add more than tha last two values as lag features when forecasting 24 steps ahead

File: tests/unit/modules/test_select.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            ({"start": 1, "stop": 4, "step": 1},
+             xr.DataArray([[2, 3, 4], [3, 4, 5], [4, 5, 6], [5, 6, 7]],
+                          dims=["time", "horizon"],
+                          coords={"time": pd.date_range('2000-01-01', freq='24H', periods=4)})),
+            ({"start": -2, "stop": 1, "step": 1},
+             xr.DataArray([[1, 2, 3], [2, 3, 4], [3, 4, 5], [4, 5, 6], [5, 6, 7]],
+                          dims=["time", "horizon"],
+                          coords={"time": pd.date_range('2000-01-03', freq='24H', periods=5)})),
+            ({"start": 2, "stop": None, "step": None},
+             xr.DataArray([[3], [4], [5], [6], [7]],
```
Comment (olineumann) [inline tests/unit/modules/test_select.py:55]: Do we really expect this here or either [3, 4, 5, 6, 7]?

File: tests/unit/modules/test_select.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                         })
+        self.select.set_params(start=3, stop=20, step=12)
+        self.assertEqual(self.select.get_params(),
+                         {
+                             "start": 3,
+                             "stop": 20,
+                             "step": 12,
+                         })
+
+    def test_transform(self):
```
Comment (olineumann) [inline tests/unit/modules/test_select.py:41]: Are there more edge cases like start=stop or step larger then array?

File: pywatts/modules/preprocessing/select.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        :rtype: xr.DataArray
+        """
+        indexes = _get_time_indexes(x)
+        if isinstance(self.start, list):
+            to_select = self.start
+        elif not self.stop is None and not self.step is None:
+            to_select = range(self.start, self.stop, self.step)
+        else:
+            to_select = [self.start]
+        r = [x.shift({index: -1 * i for index in indexes}) for i in to_select]
```
Comment (olineumann) [inline pywatts/modules/preprocessing/select.py:90]: If indexes contain only one element we may do not concat r afterwards?

File: tests/unit/modules/test_select.py (hunk 3 above)
Comment (olineumann) [reply tests/unit/modules/test_select.py:41]: Or only stop is set?

File: pywatts/modules/preprocessing/select.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        """
+        Select the desired subset from the given time series x
+
+        :param x: the input
+        :type x: xr.DataArray
+        :return: A shifted time series.
+        :rtype: xr.DataArray
+        """
+        indexes = _get_time_indexes(x)
+        if isinstance(self.start, list):
```
Comment (olineumann) [inline pywatts/modules/preprocessing/select.py:84]: Why check if self.start ist list? In the tests start always is integer.

File: pywatts/modules/preprocessing/select.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            to_select = self.start
+        elif not self.stop is None and not self.step is None:
```
Comment (olineumann) [inline pywatts/modules/preprocessing/select.py:86]: If stop is not None step has to be set? Wouldn't be step=1 suitable?

**Review by olineumann — COMMENTED** [review]
I really like the new Select Module. Just have som questions left: Mainly:
- step has no default value. Correct?
- start could be a list of integer but never tested.
- what if only start is set. is concat needed?

File: examples/example.py (hunk 1 above)
Comment (benHeid) [reply examples/example.py:43]: Yes you are right. However, I would prefer to  do this in a separate PR together with the extension of the examples. I just replicated the old behaviour with the Select Module.
In the extended examples there should be more assertions etc. for using them also as integration tests.

File: pywatts/modules/preprocessing/select.py (hunk 5 above)
Comment (benHeid) [reply pywatts/modules/preprocessing/select.py:84]: Thank you, we should add a further test which passes a list. E.g. if a forecaster aims to consider the 1, 24 and 168 hour lags.

File: tests/unit/modules/test_select.py (hunk 2 above)
Comment (benHeid) [reply tests/unit/modules/test_select.py:55]: Thank you, I change this

File: pywatts/modules/preprocessing/select.py (hunk 4 above)
Comment (benHeid) [reply pywatts/modules/preprocessing/select.py:90]: Thank you this is related to one of the comments on the testing. I change that

File: tests/unit/modules/test_select.py (hunk 3 above)
Comment (benHeid) [reply tests/unit/modules/test_select.py:41]: I add a check method that checks the type of start and if start is smaller than stop. 

The step can be larger then an array. E.g. `slice(0,2, 100000000000000)` would only take the value at position 0.
